app-antivirus/fangfrisch/files/fangfrisch-has-news.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+# vim: ts=4 sw=4 noet ft=sh
+#
+# Example script to process Fangfrisch News.
+
+declare -r MAILFROM="noreply"
+declare -r MAILTO="alice@example.com"
+declare -r SUBJECT="Fangfrisch News are available"
+
+# Pick one of the following options and uncomment the 'declare'
+# statements. Otherwise, the script will not run otherwise.
+
+# Option 1: Mutt
+#declare -r MAILAPP="mutt"
+#declare -r MAILAPP_OPT=( "-s" "$SUBJECT" "$MAILTO" )
+
+# Option 2: sendmail
+#declare -r MAILAPP="sendmail"
+#declare -r MAILAPP_OPT=( "-t" )
+#export PATH="$PATH:/usr/sbin"
+
+# Option 3: swaks
+#declare -r MAILAPP="swaks"
+#declare -r MAILAPP_OPT=( "-d" "-" "-f" "$MAILFROM" "-t" "$MAILTO" )
+
+### No changes required below this line ###
+
+set -euo pipefail
+
+die() {
+	echo >&2 "$@"
+	exit 1
+}
+
+usage() {
+	die "Usage: $(basename "$0") {directory}"
+}
+
+gen_header() {
+	cat <<EOT
+From: Fangfrisch News <$MAILFROM>
+To: $MAILTO
+Subject: $SUBJECT
+
+EOT
+# Mail header must end with an empty line!
+}
+
+declare -a NEWSITEMS=()
+
+report_news() {
+	local dir=$1 ni
+	[ -d "$dir" ] || die "$dir is not a directory"
```

Comment:
If we are going to require bash anyways, why not actually use bash? :)

```suggestion
	[[ -d $dir ]] || die "$dir is not a directory"
```

And later on too.

Bash's [[ is significantly more powerful and has better semantics than POSIX [, so it's generally a good idea to use it when you can.

Reply:
Again, I am used to writing portable scripts. Hence the quotes around "$varnames" and the single brackets [ ] for tests.